### PR TITLE
add model for js reference

### PR DESCRIPTION
**Pull request by superyyrrzz**
Compared with managed reference:
* add: optional in parameter
* change: sting -> List<string> for parameter type
* remove: generic type, attribute...

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/ItemViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            {
+                var result = new Dictionary<string, object>();
+                foreach (var item in Metadata)
+                {
+                    result[item.Key] = item.Value;
+                }
+                return result;
+            }
+        }
+    }
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/ItemViewModel.cs]: same with `Metadata`

File: src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/MemberType.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+{
+    using System;
+
+    [Serializable]
+    public enum MemberType
+    {
+        Default,
+        Toc,
+        Assembly,
+        Namespace,
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/MemberType.cs]: >Namespace [](start = 8, length = 9)

....

File: src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/ItemViewModel.cs (hunk 1 above)
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/ItemViewModel.cs]: I removed some propeties like attributes, isEii.

File: src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/ExceptionInfo.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.JavaScriptReference
+{
+    using System;
+    using YamlDotNet.Serialization;
```
Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/ExceptionInfo.cs]: >using YamlDotNet.Serialization; [](start = 4, length = 31)

group namespace as usual? `YamlDotNet.Serialization` belongs to third party dependency.

**Comment by qinezh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/PageViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        [JsonProperty("items")]
+        public List<ItemViewModel> Items { get; set; } = new List<ItemViewModel>();
+
+        [YamlMember(Alias = "references")]
+        [JsonProperty("references")]
+        public List<ReferenceViewModel> References { get; set; } = new List<ReferenceViewModel>();
+
+        [YamlMember(Alias = "shouldSkipMarkup")]
+        [JsonProperty("shouldSkipMarkup")]
+        public bool ShouldSkipMarkup { get; set; }
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/PageViewModel.cs]: >ShouldSkipMarkup [](start = 20, length = 16)

javascript also needs to skip markup for intellisence?

File: src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/MemberType.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    [Serializable]
+    public enum MemberType
+    {
+        Default,
+        Toc,
+        Package,
+        Class,
+        Interface,
+        Enum,
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/MemberType.cs]: >Enum [](start = 8, length = 4)

question: is it enum in js?

Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/MemberType.cs]: It's supported in JSDoc. see: [http://usejsdoc.org/tags-enum.html](http://usejsdoc.org/tags-enum.html)

---
In reply to: [93179738](https://github.com/dotnet/docfx/pull/1016#discussion_r93179738) [](ancestors = 93179738)

File: src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/PageViewModel.cs (hunk 4 above)
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/PageViewModel.cs]: JSDoc has the a option to choose whether to markup text. This property may be used later.

---
In reply to: [93179333](https://github.com/dotnet/docfx/pull/1016#discussion_r93179333) [](ancestors = 93179333)

File: src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/ApiParameter.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.JavaScriptReference
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/ApiParameter.cs]: ManagedReference's model is shared between build and metadata command, so it's in DataContract. I think this model can be specific for JavaScript Document.

---
In reply to: [93179522](https://github.com/dotnet/docfx/pull/1016#discussion_r93179522) [](ancestors = 93179522)

File: src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/MemberType.cs (hunk 5 above)
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.JavaScriptReference/BuildInputs/MemberType.cs]: Nice catch... although it's different from C#'s namespace, it do has this type... will add it back.

---
In reply to: [93221231](https://github.com/dotnet/docfx/pull/1016#discussion_r93221231) [](ancestors = 93221231,93191129,93179738)

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**